x-pack/plugins/cases/public/components/actions/status/use_should_disable_status.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import type { CasesUI } from '../../../../common';
+import { CaseStatuses } from '../../../../common/types/domain';
+
+import { useUserPermissions } from '../../user_actions/use_user_permissions';
+
+const nonClosedCaseStatuses = [CaseStatuses.open, CaseStatuses['in-progress']];
+
+export const useShouldDisableStatus = () => {
```

Comment:
Still need to add tests, but centralized logic for status here to simplify things